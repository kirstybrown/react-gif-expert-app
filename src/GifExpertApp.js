import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = () => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState(['The Simpsons']);

    return (
        <>
            <h2>GifExpertApp</h2>
            <p>ðŸ‘‡ðŸ‘‡ Enter a topic here (minimum 3 letters and search for "pandas" if you're feeling stressed ðŸ˜‰ ) ðŸ‘‡ðŸ‘‡</p>
            <AddCategory setCategories={ setCategories } />
            <hr />

            <ol>
                {
                    categories.map( category => (
                        <GifGrid
                            key= { category }
                            category={ category }
                        />
                    ))
                }
            </ol>
        </>
    );
};

export default GifExpertApp;
