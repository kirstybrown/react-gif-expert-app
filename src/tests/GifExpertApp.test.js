import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Tests for GifExpertApp component', () => {
  
    test('GifExpertApp should be shown correctly', () => {
      
        const wrapper = shallow( <GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('should show a list of categories', () => {
      
        const categories = ['pandas', 'The Simpsons'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categories} />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
    
    
});


