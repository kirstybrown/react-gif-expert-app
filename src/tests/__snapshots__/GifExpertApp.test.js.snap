// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifExpertApp component GifExpertApp should be shown correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <p>
    ğŸ‘‡ğŸ‘‡ Enter a topic here (minimum 3 letters and search for "pandas" if you're feeling stressed ğŸ˜‰ ) ğŸ‘‡ğŸ‘‡
  </p>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Tests for GifExpertApp component should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <p>
    ğŸ‘‡ğŸ‘‡ Enter a topic here (minimum 3 letters and search for "pandas" if you're feeling stressed ğŸ˜‰ ) ğŸ‘‡ğŸ‘‡
  </p>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="pandas"
      key="pandas"
    />
    <GifGrid
      category="The Simpsons"
      key="The Simpsons"
    />
  </ol>
</Fragment>
`;
